packages/clarity-js/src/insight/snapshot.ts

```diff
@@ -14,6 +14,16 @@ export function start(): void {
     doc.start();
     getId(document.documentElement); // Pre-discover ID for page root
     interaction.observe(document);
+
+    // observe other document nodes
+    const iframes = document.querySelectorAll("iframe");
+    for (let i = 0; i < iframes.length; i++) {
+        const iframe = iframes[i];
+        if (iframe.contentDocument) {
```

Comment:
this can throw exception if iframe is cross origin so better to wrap it in try catch to avoid console errors